helix-term/src/ui/editor.rs

```diff
@@ -1370,15 +1379,43 @@ impl Component for EditorView {
             _ => false,
         };
 
-        // -1 for commandline and -1 for bufferline
-        let mut editor_area = area.clip_bottom(1);
-        if use_bufferline {
-            editor_area = editor_area.clip_top(1);
-        }
+        let editor_area = if use_bufferline {
```

Comment:
I'm curious if there's a technical reason for changing this from mutation to shadowing, or if this is an unrelated preferential change?

Reply:
It's just a preferential change

Reply:
I strongly prefer the `mut` binding here. IMO `mut` binding are more ideomatic in rust. It allows you to `use` gr to got to all uses/assignments (instead of having to grep). The code below is also significantly longer/noisier. 

Rust is definitely more of an imperative leaning language and I have seen shadowing discouraged unless you want to change the type of a variable.

Reply:
#IMO shorter doesn't mean better, the mut version is much harder to debug because there’s no mechanically guaranteed way to look for the last update, one have perform mentally jungling instead. It will only get worse when the code grows. 

With the immutable version, gd always takes me to the last update.

Reply:
shorter is better because it makes it easier to scan code. Longer code can worth it because its more explicit and therefore conveys additional information. But here there is no additional information conveyed its simply additional syntactic noise. It's also more consistent with the rest of the codebase which is even more important.

Reply:
I agree with @pascalkuthe here. I was on the fence about it, but I too prefer the `mut` version. In addition to Pascal's points, generally I think it's not great to make a stylistic change like this just because it happens to make your text editor behave in a way you prefer.

Reply:
@pascalkuthe It is easier to scan until all the mutations do not fit into a single page. 
Mutations make code harder to understand, extend and debug (in general code maintenance). 

We should aim for better **code maintenance**, not better _code scanning_.

But I do agree that invasive code-style changes are bad unless done as a whole.